src/cider/include/cider/processor/BatchProcessorContext.h

```diff
@@ -18,23 +18,60 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+#pragma once
 
 #ifndef CIDER_BATCH_PROCESSOR_CONTEXT_H
 #define CIDER_BATCH_PROCESSOR_CONTEXT_H
 
+#include <any>
 #include <functional>
 #include <memory>
 #include <optional>
 #include <vector>
 #include "cider/CiderAllocator.h"
-
+#include "exec/nextgen/context/Batch.h"
+#include "exec/operator/join/HashTableFactory.h"
 namespace cider::exec::processor {
+using CiderJoinBaseHashTableTemplate = cider_hashtable::BaseHashTable<
+    std::any,
+    std::pair<cider::exec::nextgen::context::Batch*, int>,
```

Comment:
can we introduce  a new struct 'BatchOffset' to wrap the batch pointer and the batch row index?

Reply:
good idea. Let me update it